vector/src/main/java/im/vector/app/features/spaces/manage/SpaceManageRoomsFragment.kt

```diff
@@ -0,0 +1,200 @@
+import androidx.transition.TransitionManager
+import com.airbnb.mvrx.Loading
+import com.airbnb.mvrx.activityViewModel
+import com.airbnb.mvrx.fragmentViewModel
+import com.airbnb.mvrx.withState
+import com.jakewharton.rxbinding3.appcompat.queryTextChanges
+import im.vector.app.R
+import im.vector.app.core.extensions.cleanup
+import im.vector.app.core.extensions.configureWith
+import im.vector.app.core.platform.OnBackPressed
+import im.vector.app.core.platform.VectorBaseFragment
+import im.vector.app.core.utils.toast
+import im.vector.app.databinding.FragmentSpaceAddRoomsBinding
+import io.reactivex.rxkotlin.subscribeBy
+import org.matrix.android.sdk.api.session.room.model.SpaceChildInfo
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class SpaceManageRoomsFragment @Inject constructor(
+        private val viewModelFactory: SpaceManageRoomsViewModel.Factory,
+        private val epoxyController: SpaceManageRoomsController
+) : VectorBaseFragment<FragmentSpaceAddRoomsBinding>(),
+        SpaceManageRoomsViewModel.Factory,
+        OnBackPressed,
+        SpaceManageRoomsController.Listener,
+        Callback {
+
+    private val viewModel by fragmentViewModel(SpaceManageRoomsViewModel::class)
+    private val sharedViewModel: SpaceManageSharedViewModel by activityViewModel()
+
+    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?) = FragmentSpaceAddRoomsBinding.inflate(inflater)
+
+    override fun onBackPressed(toolbarButton: Boolean): Boolean {
+        parentFragmentManager.popBackStack()
+        return true
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupToolbar(views.addRoomToSpaceToolbar)
+        views.appBarTitle.text = getString(R.string.space_manage_rooms_and_spaces)
+        views.createNewRoom.isVisible = false
+        epoxyController.listener = this
+        views.roomList.configureWith(epoxyController, hasFixedSize = true, showDivider = true)
+
+        views.publicRoomsFilter.queryTextChanges()
+                .debounce(200, TimeUnit.MILLISECONDS)
+                .subscribeBy {
+                    viewModel.handle(SpaceManageRoomViewAction.UpdateFilter(it.toString()))
+                }
+                .disposeOnDestroyView()
+
+        viewModel.selectSubscribe(SpaceManageRoomViewState::actionState) { actionState ->
+            when (actionState) {
+                is Loading -> {
+                    sharedViewModel.handle(SpaceManagedSharedAction.ShowLoading)
+                }
+                else       -> {
+                    sharedViewModel.handle(SpaceManagedSharedAction.HideLoading)
+                }
+            }
+        }
+
+        viewModel.observeViewEvents {
+            when (it) {
+                is SpaceManageRoomViewEvents.BulkActionFailure -> {
+                    it.errorList.forEach {
+                        vectorBaseActivity.toast(it)
```

Comment:
wow, lot's of toast could be displayed :/

Reply:
yes, not sure what to do TBH :/
Will see how to improve

Reply:
thx